### ISSUE TITLE
Union of sets of literals cannot be assigned to an optional set of literals

### ISSUE THREAD

**Issue by ipkiss42** [microsoft/pyright#1922]
Note: I'm not sure if this is related to #1901 or not, but it looks very similar.

Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
The union of two sets of literals cannot be passed to a function expecting an optional set of those literals. I think the code snippet below will be clearer :)

**To Reproduce**
With this code:
```python
from typing import Optional, Set
from typing import Literal

FooOrBar = Literal["foo", "bar"]

def dummy(values: Optional[Set[FooOrBar]]) -> None:
    pass

def combine(set1: Set[FooOrBar], set2: Set[FooOrBar]) -> None:
    dummy(values=set1 | set2)
```

Pyright wrongly complains about typing:
```
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/pyright1/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/pyright1/foo.py
  /tmp/pyright1/foo.py:10:18 - error: Argument of type "Set[str]" cannot be assigned to parameter "values" of type "Set[FooOrBar] | None" in function "dummy"
    Type "Set[str]" cannot be assigned to type "Set[FooOrBar] | None"
      Type cannot be assigned to type "None"
        TypeVar "_T@set" is invariant
          Type "str" cannot be assigned to type "FooOrBar"
            "str" cannot be assigned to type "Literal['foo']" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
Completed in 0.651sec
```

**Expected behavior**
No issue reported

**VS Code extension or command-line**
Command-line (Pyright 1.1.144)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.145, which I just published. It will also be included in the next release of pylance.
